### DISCUSSION TITLE
[Run issue]:TsDNAELSEPAElasticModel.cc:190:28: error: ‘G4FindDataDir’ was not declared in this scope

### DISCUSSION THREAD

**Discussion by nknanfeng** [topas-nbio/TOPAS-nBio#22] [Run-Problem] [answered]
### What was the issue?

I encountered the following function noit found issue when excuting the make command to compile.How should I resolve this?Did I miss any header files?

`/home/tdw/topas/build/extensions/TsDNAELSEPAElasticModel.cc: In member function ‘virtual void TsDNAELSEPAElasticModel::Initialise(const G4ParticleDefinition*, const G4DataVector&)’:
/home/tdw/topas/build/extensions/TsDNAELSEPAElasticModel.cc:190:28: error: ‘G4FindDataDir’ was not declared in this scope
  190 |         const char *path = G4FindDataDir("G4LEDATA");
      |                            ^~~~~~~~~~~~~
/home/tdw/topas/build/extensions/TsDNAELSEPAElasticModel.cc:266:28: error: ‘G4FindDataDir’ was not declared in this scope
  266 |         const char *path = G4FindDataDir("G4LEDATA");
      |`

### TOPAS-nBio category

Geometry

### Version

3.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment]
G4FindDataDir function is available from Geant4 v11.1, so the TOPAS version you're using is outdated. Please update it from here https://github.com/OpenTOPAS/OpenTOPAS

**Comment by nknanfeng** [comment]
Thanks for your answers. I am using Geant4 version v11.2.1, and I noticed that the G4FindDataDir.hh file is not included in the Geant4Headers directory in TOPAS. Does this have any impact? I didn't encounter any issues when compiling TOPAS-nBio-v2.0.Which version of Geant4 are you using?

**Comment by JoseRamosMendez** [comment, marked as answer]
Check [here](https://nbio.readthedocs.io/en/latest/) for details. In short, for **TOPAS-nBio 3.0** please use the TOPAS **version OpenTOPAS_4.0**. You can download it at here https://github.com/OpenTOPAS/OpenTOPAS.git. In that repository you'll find detailed installation instructions; it uses **Geant4 v11.1.3.**

**Comment by nknanfeng** [comment]
Thanks, you are my hero.

**Comment by nknanfeng** [comment]
Dear&nbsp;JoseRamosMendez,

I hope this message finds you well.

I am currently working with the /examples/geometry/nucleusModel/Nucleus.txt example from TOPAS-nBio. I am encountering issues with running this example on my personal computer and am seeking guidance on the required system specifications for optimal performance.

Could you please provide the following details:

Computer Configuration: What are the recommended hardware specifications (CPU, RAM, GPU, etc.) for running this example efficiently?

Performance Details: What was the computer configuration used during your experiments, and how long did it take to complete the task?

Understanding these requirements will greatly assist me in troubleshooting the performance issues I am experiencing and achieving the desired output.

Thank you for your assistance. I look forward to your response.

Best regards,



南枫
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "topas-nbio/TOPAS-nBio"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月29日(星期一) 凌晨3:43
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [topas-nbio/TOPAS-nBio] [Run issue]:TsDNAELSEPAElasticModel.cc:190:28: error: ‘G4FindDataDir’ was not declared in this scope (Discussion #22)





 
Check here for details. In short, for TOPAS-nBio 3.0 please use the TOPAS version OpenTOPAS_4.0. You can download it at here https://github.com/OpenTOPAS/OpenTOPAS.git. In that repository you'll find detailed installation instructions; it uses Geant4 v11.1.3.
